hosts/github.go

```diff
@@ -33,9 +33,22 @@ func NewGithubClient(m manager.Manager) *Github {
 		&oauth2.Token{AccessToken: options.GetAccessToken(m.Opts)},
 	)
 
+	var githubClient *github.Client
+	httpClient := oauth2.NewClient(ctx, token)
+
+	if m.Opts.BaseURL == "" {
+		githubClient = github.NewClient(httpClient)
+	} else {
+		var err error
+		githubClient, err = github.NewEnterpriseClient(m.Opts.BaseURL, m.Opts.BaseURL, httpClient)
+		if err != nil {
+			log.Error(err)
+		}
+	}
+
 	return &Github{
```

Comment:
This all looks good but let's add a nil check here https://github.com/zricethezav/gitleaks/blob/master/hosts/host.go#L31. Something like 
```
	if host == nil {
		return fmt.Errorf("unable to create %s client", getHost(m.Opts.Host))
	}
```

Reply:
Or we could change the function signature to return an error

Reply:
Yeah, makes a lot more sense to return the errors.